pkg/entities/types.go

```diff
@@ -5272,34 +5272,10 @@ type DashboardTableWidgetConfiguration struct {
 	NRQLQueries []DashboardWidgetNRQLQuery `json:"nrqlQueries,omitempty"`
 }
 
-// DashboardVariable - Definition of a variable that is local to this dashboard. Variables are placeholders for dynamic values in widget NRQLs.
-type DashboardVariable struct {
-	// [DEPRECATED] Default value for this variable. The actual value to be used will depend on the type.
-	DefaultValue DashboardVariableDefaultValue `json:"defaultValue,omitempty"`
-	// Default values for this variable. The actual value to be used will depend on the type.
-	DefaultValues []DashboardVariableDefaultItem `json:"defaultValues,omitempty"`
-	// Indicates whether this variable supports multiple selection or not. Only applies to variables of type NRQL or ENUM.
-	IsMultiSelection bool `json:"isMultiSelection,omitempty"`
-	// List of possible values for variables of type ENUM.
-	Items []DashboardVariableEnumItem `json:"items,omitempty"`
-	// Configuration for variables of type NRQL.
-	NRQLQuery DashboardVariableNRQLQuery `json:"nrqlQuery,omitempty"`
-	// Variable identifier.
-	Name string `json:"name,omitempty"`
-	// Options applied to the variable.
-	Options DashboardVariableOptions `json:"options,omitempty"`
-	// Indicates the strategy to apply when replacing a variable in a NRQL query.
-	ReplacementStrategy DashboardVariableReplacementStrategy `json:"replacementStrategy,omitempty"`
-	// Human-friendly display string for this variable.
-	Title string `json:"title,omitempty"`
-	// Specifies the data type of the variable and where its possible values may come from.
-	Type DashboardVariableType `json:"type,omitempty"`
-}
-
 // DashboardVariableDefaultItem - Represents a possible default value item.
 type DashboardVariableDefaultItem struct {
 	// The value of this default item.
-	Value DashboardVariableDefaultValue `json:"value,omitempty"`
+	Value *DashboardVariableDefaultValue `json:"value,omitempty"`
```

Comment:
@sanderblue if my understanding I've shared in the previous comment ⬆️ is true, I think all of these changes are fine 👍 I've compared these with the changes in 2.37.1 of the client and they are entirely in sync. 

However, I see the section highlighted in **this** comment, `*DashboardVariableDefaultValue` inside `DashboardVariableDefaultItem` as the only exception (the only change that does not exist in versions of the client prior to the entities cleanup PR merge). 

Now that `DashboardVariableDefaultValue` is under Tutone scope (is being generated as `*DashboardVariableDefaultValue`), and since DashboardVariableDefaultItem contains `DashboardVariableDefaultValue`, this is also being changed to `*DashboardVariableDefaultValue`. 

Unfortunately, this was not the case with in versions of the client before the `entities` cleanup PR was merged. It looks like a manual change was made to `*DashboardVariableDefaultValue` and `*[]DashboardVariableDefaultItem` in `DashboardVariable` to be taken as a pointer, but because the change to  `*DashboardVariableDefaultValue` was not Tutone driven, this did not propagate into `DashboardVariableDefaultValue` inside `DashboardVariableDefaultItem`, as a result of which this is what `DashboardVariableDefaultItem` looks like in v2.37.1 of the client.

https://github.com/newrelic/newrelic-client-go/blob/8a5c6dc32520fda24a39ad120e474c91edad2bbe/pkg/entities/types.go#L7976-L7980

https://github.com/newrelic/newrelic-client-go/blob/8a5c6dc32520fda24a39ad120e474c91edad2bbe/pkg/entities/types.go#L8005-L8009

This should absolutely be the right change, but I believe with this we might need additional changes in the Terraform Provider, so just wanted to post this as a thought.

Reply:
I think we're good to go for this PR. Overall, what you said is correct with regard to how I made these changes with our tooling. I will create a followup PR if anything else comes up in the provider.